Dockerfile

```
@@ -1,2 +1,11 @@
 FROM golang:1.17.0-alpine3.14
 
+ENV PATH /usr/local/git/bin:$PATH
+
+RUN apk add --no-cache git
+
+WORKDIR PATH
+
+COPY ["./go.mod /go.mod","./main.go /main.go",]
```

```suggestion
COPY ["./go.mod","./main.go", "/"]
```

COPYを1つのレイヤーで対応する場合は、このような書き方になります。
最後の引数の場所にに手前の2つがCOPYされます。

---

元々の記載で何故動いたかは不明です。
ちなみに、こちらのwindows環境でも動きませんでした。